### PR TITLE
Set source-type flag required

### LINKED ISSUE #10960: [mesheryctl] Issue Converting Helm Chart to Kubernetes Manifest when Importing Design via URL
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

The error occurs importing design patterns via URL with `mesheryctl pattern import`. The issue arises during Helm Chart to Kubernetes Manifest conversion. Importing succeeds with local files.

File link: https://raw.githubusercontent.com/meshery/meshery.io/master/catalog/c40ae08b-deaa-4fc5-a1e8-8081907b64a3.yaml
### Screenshots/Logs

Screenshot from 2024-05-15 11-32-33

### Environment

- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by lekaf974**
**Notes for Reviewers**

This PR fixes #10960 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 

Due to non provided source type, the server returned the error. To avoid this behavior the -s/--source-type flag is now required

new behavior
```bash
[xxx@xxx mesheryctl]$ mesheryctl pattern import -f https://raw.githubusercontent.com/meshery/meshery.io/master/catalog/c40ae08b-deaa-4fc5-a1e8-8081907b64a3.yaml
source-type not provided
Error: No manifest source-type detected
```

### CODE REVIEW

**Comment by leecalcote** [comment]
@Yashsharma1911 @MUzairS15 does Meshery Server still not auto-detect design source type?

File: mesheryctl/internal/cli/root/pattern/error.go [hunk 1]
```diff
@@ -85,6 +86,14 @@ func ErrPatternManifest() error {
 		[]string{"Provide the path to the pattern manifest. \n\n%v", errPatternMsg})
 }
 
+func ErrPatternSourceType() error {
+	return errors.New(ErrPatternSourceTypeCode, errors.Alert,
+		[]string{"No source-type detected"},
+		[]string{"No manifest source-type detected"},
+		[]string{"Manifest source-type not provided"},
+		[]string{"Provide the source-type to the pattern manifest. \n\n%v", errPatternMsg})
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/pattern/error.go:94]:
```suggestion
		[]string{"Provide one of the supported source type for the design to import. \n\n%v", errPatternMsg})
```

**Review by MUzairS15 — APPROVED** [review]

File: mesheryctl/internal/cli/root/pattern/error.go [hunk 2]
```diff
@@ -85,6 +86,14 @@ func ErrPatternManifest() error {
 		[]string{"Provide the path to the pattern manifest. \n\n%v", errPatternMsg})
 }
 
+func ErrPatternSourceType() error {
+	return errors.New(ErrPatternSourceTypeCode, errors.Alert,
+		[]string{"Source type for the design to import not specified"},
+		[]string{"Empty  source type
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/pattern/error.go:92]:
```suggestion
		[]string{"Empty  source type detected
```

File: mesheryctl/internal/cli/root/pattern/error.go [hunk 3]
```diff
@@ -85,6 +86,14 @@ func ErrPatternManifest() error {
...
+		[]string{"Manifest source-type not provided"},
```
Comment (MUzairS15) [inline mesheryctl/internal/cli/root/pattern/error.go:93]:
```suggestion
		[]string{"Design source type not provided"},
```

**Comment by MUzairS15** [comment]
@lekaf974 please don’t assign new error code value manually.
https://docs.meshery.io/project/contributing/contributing-error

Just define the new error and the utility will take care of it.
Use  `make error-util` from mesheryctl

**Comment by MUzairS15** [comment]
Will you address the lint checks?

**Comment by lekaf974** [comment]
updated but still having issue in file not related to this PR
```bash
pkg/utils/model.go:119:27: Error return value of `mcv.UpdateModelDefinition` is not checked (errcheck)
	mcv.UpdateModelDefinition(&model)
```

**Comment by MUzairS15** [comment]
thanks @lekaf974

**Pull request merged**